datadog-checks-base/datadog_checks/stubs/aggregator.py

```diff
@@ -0,0 +1,89 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2018 Datadog, Inc.
+from collections import defaultdict, namedtuple
+
+MetricStub = namedtuple('MetricStub', 'name type value tags hostname')
+
+
+class AggregatorStub(object):
+    """
+    Mainly used for unit testing checks, this stub makes possible to execute
+    a check without a running Agent.
+    """
+    # Replicate the Enum we have on the Agent
+    GAUGE, RATE, COUNT, MONOTONIC_COUNT, COUNTER, HISTOGRAM, HISTORATE = range(7)
+
+    def __init__(self):
+        self.reset()
+
+    def submit_metric(self, check, check_id, mtype, name, value, tags, hostname):
+        self._metrics[name].append(MetricStub(name, mtype, value, tags, hostname))
+
+    def submit_service_check(self, *args, **kwargs):
+        pass
+
+    def submit_event(self, *args, **kwargs):
+        pass
+
+    def metrics(self, name):
+        """
+        Return the metrics received under the given name
+        """
+        return self._metrics.get(name, [])
+
+    def assert_metric(self, name, value=None, tags=None, count=None, at_least=1,
```

Comment:
We should have another method to test value in range: `assert_metric_in_range` ?
That may come in another PR.

Reply:
I expect this class to grow quite a bit as long as we move more checks to the new testing strategy, so I'd keep these additions for a future PR with maybe a real use case. 
On the other side, I think we should keep track of your comments, maybe on a GH issue?

Reply:
OK !